Functions/InputOutput/xASL_io_ReadDataPar.m

```diff
@@ -147,37 +147,56 @@
 	else
 		x = jsonData;
 	end
-    
-	sFields = fieldnames(x);
-	for n=1:size(sFields,1)
-		% Check if the current field is valid (char, numeric value, structure or cell array)
-		if ~(ischar(x.(sFields{n})) || isstruct(x.(sFields{n})) || isnumeric(x.(sFields{n})) || islogical(x.(sFields{n})) || iscell(x.(sFields{n})))
-			fprintf('\n%s\n',char(sFields{n}));
-			warning('JSON field type could be invalid...');
+	if isfield(x,'StudyPars')
+		bStudyPars = 1;
+		nContext = length(x.StudyPars);
+	else
+		bStudyPars = 0;
+		nContext = 1;
+	end
+	
+	for iSubfield = 1:nContext
+		if bStudyPars
+			subx = x.StudyPars{iSubfield};
+		else
+			subx = x;
 		end
-		if strcmpi(sFields{n},'group') % Convert group fields to correct Matlab cell arrays
-			% Generate new Matlab cell array
-			x.group{str2num(strrep(sFields{n},'group',''))} = x.(sFields{n});
-			% Remove old field
-			x = rmfield(x,sFields{n});
-		elseif strcmpi(sFields{n},'LOAD') % Handle load commands
-			loadPaths = fieldnames(x.(sFields{n}));
-			for m=1:size(loadPaths,1)
-				% Don't forget to use \\ instead of \ in your paths
-				if exist(x.(sFields{n}).(loadPaths{m}),'file')
-					load(fullfile(x.(sFields{n}).(loadPaths{m})));
+		sFields = fieldnames(subx);
+		for n=1:size(sFields,1)
+			% Check if the current field is valid (char, numeric value, structure or cell array)
+			if ~(ischar(subx.(sFields{n})) || isstruct(subx.(sFields{n})) || isnumeric(subx.(sFields{n})) || islogical(subx.(sFields{n})) || iscell(subx.(sFields{n})))
+				fprintf('\n%s\n',char(sFields{n}));
+				warning('JSON field type could be invalid...');
+			end
+			if strcmpi(sFields{n},'group') % Convert group fields to correct Matlab cell arrays
+				% Generate new Matlab cell array
+				subx.group{str2num(strrep(sFields{n},'group',''))} = subx.(sFields{n});
+				% Remove old field
+				subx = rmfield(subx,sFields{n});
+			elseif strcmpi(sFields{n},'LOAD') % Handle load commands
+				loadPaths = fieldnames(subx.(sFields{n}));
+				for m=1:size(loadPaths,1)
+					% Don't forget to use \\ instead of \ in your paths
+					if exist(subx.(sFields{n}).(loadPaths{m}),'file')
+						load(fullfile(subx.(sFields{n}).(loadPaths{m})));
+					end
 				end
 			end
 		end
+		
+		% Convert strings containing numbers to number
+		subx = xASL_io_ReadDataPar_FixFields(subx);
```

Comment:
note that this is not on subject level but on SubjVisitSess level

Reply:
Not sure if I understand. This is once per studyPar = once per the whole import.

Reply:
ah OK I thought studyPar is now repeated for each possible combination. So for each subject/visit/session level?

Reply:
this function checks and cleans studyPar (saved in x). If there are multiple studyPars, then we go one by one, save each to `subfieldX`, clean it using the normal procedure, and then put it back cleaned to the list of studyPars.

We don't repeat for all combinations of subject/visit/session, we simply go through the list of studyPars.